src/System.Device.Gpio/System/Device/I2c/UnixI2cBus.cs

```diff
@@ -152,12 +172,12 @@ internal unsafe void WriteRead(int deviceAddress, ReadOnlySpan<byte> writeBuffer
             {
                 fixed (byte* readBufferPointer = readBuffer)
                 {
-                    WriteReadCore((ushort)deviceAddress, writeBufferPointer, readBufferPointer, (ushort)writeBuffer.Length, (ushort)readBuffer.Length);
+                    WriteReadCore((ushort)deviceAddress, writeBufferPointer, readBufferPointer, (ushort)writeBuffer.Length, (ushort)readBuffer.Length, true);
                 }
             }
         }
 
-        protected virtual unsafe void WriteReadCore(ushort deviceAddress, byte* writeBuffer, byte* readBuffer, ushort writeBufferLength, ushort readBufferLength)
+        protected virtual unsafe bool WriteReadCore(ushort deviceAddress, byte* writeBuffer, byte* readBuffer, ushort writeBufferLength, ushort readBufferLength, bool throwExceptionOnIOError)
```

Comment:
Technically, this is a breaking change, but I guess there won't be many derived classes that are affected.